docs/yul_language/yul.rst

```diff
@@ -0,0 +1,14 @@
+Overview of Yul
+===============
+
+.. _yul_section:
+
+Yul, also know as EVM assembly, is a low level language for creating smart contracts and for providing more
+control over the execution environment when using Solidity as the primary language. Although it enables
+more possibilities to manage memory, using Yul does not imply a performance improvement. In Solang,
+all Yul constructs are processed using the same pipeline as Solidity.
+
+The support for Yul is only partial. We support all statements, except for the switch-block. In addition,
+some Yul builtins are not yet implemented, but we are trying to improve compatibility. In the following sections,
```

Comment:
do not give the impression that all builtins will be implemented.

remove `, but we are trying to improve compatibility`, this is implied and rather than focus on what we haven't done, it's better to focus on what we have done.